### ISSUE TITLE
[Feature Request] Tombstones in checkpoint

### ISSUE THREAD

**Issue by felipepessoto** [delta-io/delta#3923]
#### Which Delta project/connector is this regarding?
- [X] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

### Overview
I'd like to confirm what is the current usage of tombstones in the checkpoint files. I couldn't find any references to it in the code.

If they are not used, I'd suggest having an option to don't include it in the checkpoint, since is some scenarios like when running MERGE command could rewrite all the files, and the checkpoint become huge and most of the data are tombstones.

### Motivation
Tombstones are likely only wasting space and causing performance issues.

### Willingness to contribute

The Delta Lake Community encourages new feature contributions. Would you or another member of your organization be willing to contribute an implementation of this feature?

- [ ] Yes. I can contribute this feature independently.
- [ ] Yes. I would be willing to contribute this feature with guidance from the Delta Lake community.
- [X] No. I cannot contribute this feature at this time.

**Comment by felipepessoto** [comment]
@raveeram-db, later I found they are used for VACUUM, to know the deletion time. Can I close this? Thanks

**Comment by raveeram-db** [comment]
Yes, please go ahead. We're also still exploring ideas around potentially optimizing this and making checkpoints faster. We'll open an issue then.

**Comment by felipepessoto** [comment]
@raveeram-db, not related to this, but do you think somebody could help reviewing FSCK PRs?

https://github.com/delta-io/delta/pull/3520
https://github.com/delta-io/delta/pull/3499
